### ISSUE TITLE
Sync button not shown, even when there is space for it

### ISSUE THREAD

**Issue by sphh** [kitsunyan/foxy-droid#51]
I love your F-Droid alternative! Thanks.

I noticed, that the `Sync` button is not shown on my phone as mentioned in https://github.com/kitsunyan/foxy-droid/issues/46#issuecomment-674518831 and https://github.com/kitsunyan/foxy-droid/issues/30#issuecomment-674041115. I admit, that my screen is rather small, but I would have thought, that there is still some space left for the sync button:
![Screenshot_20200829-155006](https://user-images.githubusercontent.com/4856462/91638502-94c39780-e9ff-11ea-9d8b-821e47447f27.png)
There should also be enough space in the screenshot shown in https://github.com/kitsunyan/foxy-droid/issues/30#issuecomment-674041115 …

If it is not feasible to "squeeze" the `Sync` button into the available space, I would suggest to put the `Sync` button there and move the `Sorting order` button to the menu instead. The rational is, that the `Sync` button is used more often than the `Sorting order` button.

Foxy-Droid version: 1.3
Screen Size: 240x432px
Dp: 320x576dp

**Comment by kitsunyan** [comment]
> The rational is, that the `Sync` button is used more often than the `Sorting order` button.

For me it isn't. I almost never use "sync" button.

**Comment by kitsunyan** [comment]
> that there is still some space left for the sync button:

It's reserved for search input. It doesn't have enough space if I place more buttons there.

**Comment by sphh** [comment]
Can you please outline your workflow for us? How do you sync?

I open the app, which opens on the `AVAILABLE` tab. Nearly all of the time I reach immediatley for the `Sync` button (or in my case the menu and then `Sync`) because that is why I open the app: I want to know if there are any updates. I very, very rarely open `Foxy Droid` to search for an app …

**Comment by kitsunyan** [comment]
> Can you please outline your workflow for us? How do you sync?

I use "sync repositories automatically" and wait for a notification. It checks 2 times a day. Even if you have notifications disabled, it will sync and "update" tab will have new data.

**Comment by sphh** [comment]
I also have `Sync repositories automatically` enabled, but `Foxy Droid` never sync'ed the repositories automatically (and hence I never got any notification). I always had to do a manual update (and then got the notification). Hence my questions:

1. At what times does `Foxy Droid` update the repository?
2. Can the powersave settings disturb the automatic update?

If 2. is the culprit, I have two possible settings, which I can change (on Android 8.1):
1. `Settings > Battery > ⋮ >  Battery optimisation` and set `Foxy Droid` to `Not optimized`
2. `Settings > Smart Assistant >  Power save manager` and put `Foxy Droid` into the white list
Which one would enable `Foxy Droid` to automatically sync the repositories?

**Comment by kitsunyan** [comment]
> At what times does `Foxy Droid` update the repository?

I can't say for sure anything but "about 2 times in a day". This is how JobScheduler works.

> Can the powersave settings disturb the automatic update?

No, unless your device manufacturer has its own power saving "features".

> Which one would enable Foxy Droid to automatically sync the repositories?

You can try to experiment yourself. The first option shouldn't prevent from syncing, because it's a standard power saving feature in Android. I'm not sure about the smart assistant, I don't know what is it.

You can also enter `dumpsys jobscheduler` in a terminal emulator (you can use adb for example) and check the schedules and logs.

**Comment by sphh** [comment]
`dumpsys jobscheduler` showed entries for `Foxy Droid`, but it still took more than 24 hours to update and to show a notification (I checked with `F-Droid` that there is an update available). I also whitelisted `Foxy Droid` in the `Settings > Smart Assistant > Power save manager`, because I knew from other apps, which use the scheduler, that this is required.

Looks like it is working (with a very loose interpretation of "about 2 times a day" …), so I close this issue. Thank you for your help and support!
